### PR TITLE
Add correct permissions for monitoring service and make its creation optional

### LINKED ISSUE #41: missing permissions for creating services
**Describe the bug**
I've deployed the secret-generator according to the example manifests given in the `deploy` dir in this repo, and its complaining about missing rbac permissions for creating services.
This raises 2 questions:
1. If this permission is really needed, why is it not mentioned in the example?
2. Why is it needed? Why does a secret-generator needs to create service object(s) in the cluster?

**To Reproduce**
Deploy the secret manager according to the example given in `deploy` dir.

**Expected behavior**
The container should start without any errors

**Environment:**
 - Kubernetes version: 1.18.16
 - kubernetes-secret-generator version: 3.2.0

**Additional context**
```
kubernetes-secret-generator {"level":"info","ts":1617775252.2373734,"logger":"cmd","msg":"Could not create metrics Service","error":"failed to create or get service for metrics: services is forbidden: User \"system:serviceaccount:kube-system:kubernetes-secret-generator\" cannot create resource \"services\" in API group \"\" in the namespace \"kube-system\""}
```

### PR DESCRIPTION
**Pull request by YannikBramkamp**
Closes #41

### CODE REVIEW

**Comment by martin-helmich** [comment]
@YannikBramkamp This one is still marked as "Draft" and is getting stale by now. Is this ready for review, or does it still need some ❤️?

**Comment by YannikBramkamp** [comment]
> @YannikBramkamp This one is still marked as "Draft" and is getting stale by now. Is this ready for review, or does it still need some heart?

@martin-helmich I figured I'd wait for #38 to be merged, since I'll preobably have to make some changes to this pr.

**Comment by YannikBramkamp** [comment]
This should be ready now @martin-helmich

File: Makefile [hunk 1]
```diff
@@ -21,6 +31,16 @@ uninstall: ## Uninstall all that all performed in the $ make install
 	@echo ....... Deleting Operator .......
 	kubectl delete -f deploy/operator.yaml -n ${NAMESPACE}
 
+.PHONY: uninstallwithmonitoring
+uninstallwithmonitoring: ## Uninstall all that all performed in the $ make installwithmonitoring
+	@echo ....... Uninstalling .......
+	@echo ....... Deleting Rules and Service Account .......
+	kubectl delete -f deploy/role.yaml -n ${NAMESPACE}
```
Comment (hensur) [inline Makefile:38]:
```suggestion
	kubectl apply -f deploy/role_with_service_permissions.yaml -n ${NAMESPACE}
```

File: README.md [hunk 2]
```diff
@@ -19,6 +19,8 @@ You might want to take a look a the [values.yaml](deploy/helm-chart/kubernetes-s
 
 - `watchNamespace` defines, which namespaces should be watched for secret objects.
 
+- `useMetricsService` toggles whether the operator should provide a service for metrics monitoring by Prometheus. If this is set to true, the operator will start with additional permissions, namely `get` permissions for `replicasets` and `deployments` in the apiGroup `apps`, as well as `create` permissions for `services` in the apiGroup `""` and create the needed services during startup.
```
Comment (hensur) [inline README.md:22]: "the apiGroup `""`" looks a bit strange, I would probably just call it the `core` apiGroup or completely omit the group.

**Review by hensur — CHANGES_REQUESTED** [review]

**Review by hensur — APPROVED** [review]

**Pull request merged**